### PR TITLE
fix(external docs): Add documentation for missing VRL functions

### PR DESCRIPTION
**Pull request by lucperkins**
This PR adds missing CUE docs for the `assert_eq` VRL function. It also re-factors the `vrl/examples` shortcode to be more user friendly.

### CODE REVIEW

File: docs/cue/reference/remap/functions/assert_eq.cue [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package metadata
```
Comment (jszwedko) [inline docs/cue/reference/remap/functions/assert_eq.cue:1]: Thanks for adding this. I see I missed it in my initial PR.

File: docs/cue/reference/remap/functions/assert_eq.cue [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+remap: functions: assert_eq: {
+	category: "Debug"
+
+	description: """
+		Asserts that two expressions, `left` and `right`, have the same value. If the values are
+		equivalent, `true` is returned, otherwise `false` is returned. You can specify an optional
+		`message` to append to the error output.
```
Comment (jszwedko) [inline docs/cue/reference/remap/functions/assert_eq.cue:9]: I think this should more closely match the description in `assert` as it is essentially the same as `assert(left == right)`. Something like:

```suggestion
Asserts that two expressions, `left` and `right`, have the same value. The program is aborted with the `message` if they are unequal.
```

And then have a similar `notice`:

```
		The `assert_eq` function should be used in a standalone fashion and only when you want to abort the program. You should avoid it in logical expressions and other situations in which you want the program to continue if the condition evaluates to `false`.
```

File: docs/cue/reference/remap/functions/assert_eq.cue [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package metadata
```
Comment (jszwedko) [inline docs/cue/reference/remap/functions/assert_eq.cue:1]: Could we add the same `notice` we have on `assert` here?

**Review by jszwedko — COMMENTED** [review]
Just one comment, otherwise this looks good!

File: docs/cue/reference/remap/functions/assert_eq.cue (hunk 3 above)
Comment (lucperkins) [reply docs/cue/reference/remap/functions/assert_eq.cue:1]: Ah, derp. Totally meant to do that. Fixed.

**Comment by lucperkins** [comment]
@jszwedko FYI these Netlify build errors are spurious. I'm in touch with their support.

**Review by jszwedko — APPROVED** [review]

**Pull request merged**